src/librustc_driver/driver.rs

```diff
@@ -833,10 +833,6 @@ pub fn phase_2_configure_and_expand<F>(sess: &Session,
         })
     })?;
 
-    time(time_passes,
-         "early lint checks",
-         || lint::check_ast_crate(sess, &krate));
-
```

Comment:
Yes, after https://github.com/rust-lang/rust/pull/43522 you can't report late lints without `tcx`, and you can't report early lints after `lint::check_ast_crate` is run. So you can't report lints during AST -> HIR lowering at all.
Thus, I had to move `lint::check_ast_crate` slightly further in the pipeline, to after lowering.